frontend/src/concepts/pipelines/content/StartingStatusModal.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+import {
+  Modal,
+  ModalBody,
+  ModalHeader,
+  ModalFooter,
+  Spinner,
+  Stack,
+  StackItem,
+  Tab,
+  Tabs,
+  TabTitleText,
+  Panel,
+  PanelMain,
+  List,
+  ListItem,
+  Button,
+} from '@patternfly/react-core';
+import { usePipelinesAPI } from '#~/concepts/pipelines/context';
+import { getStatusFromCondition } from '#~/concepts/pipelines/content/utils.tsx';
+import K8sStatusIcon from '#~/concepts/pipelines/content/K8sStatusIcon.tsx';
+import { K8sCondition } from '#~/k8sTypes.ts';
+
+const PROGRESS_TAB = 'Progress';
+const EVENT_LOG_TAB = 'Events log';
+
+const notReadySection = (
+  <div>
+    <p style={{ textAlign: 'center' }} />
+    <div>
+      This may take a while. You can close this modal and continue using the application. The
+      pipeline server will be available when initialization is complete.
+    </div>
+  </div>
+);
+
+const CONTENT_HEIGHT = 470;
+
+type StartingStatusModalProps = {
+  onClose: () => void;
+  conditionLog: K8sCondition[];
+};
+
+// because you can close and re-open this modal; and the 'status' tab only needs to show the current conditions;
+// but the 'event log' tab should show everything (from before the user opens the modal)
+// need to get the events from the parent that already/always exists.
+// (this component only exists when it is open): eslint rules do not allow this component to exist if it is not open
+const StartingStatusModal: React.FC<StartingStatusModalProps> = ({ onClose, conditionLog }) => {
+  const { pipelinesServer } = usePipelinesAPI();
+  const [activeTab, setActiveTab] = React.useState<string>(PROGRESS_TAB);
+  const isServerReadyAndCompletelyDone = pipelinesServer.crStatus?.conditions?.some(
+    (c) => c.type === 'Ready' && c.status === 'True',
+  );
+
+  const spinner = (
+    <div className="pf-v6-u-display-flex">
```

Comment:
dont use classnames

reason is that they are tied to Patternfly 6. when we upgrade to the next patternfly, someone has to go through every one of these and fix them. better to not use them

Reply:
no use the `Flex` component

Reply:
generally we always want to use the patternfly components and props as much as possible bc classnames are a pain to migrate on upgrade